driver-scala/src/main/scala/org/mongodb/scala/model/search/SearchCollector.scala

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.mongodb.scala.model.search
+
+import com.mongodb.annotations.Beta
+import com.mongodb.client.model.search.{ SearchCollector => JSearchCollector }
+import org.mongodb.scala.bson.conversions.Bson
+import org.mongodb.scala.model.Projections
+
+import scala.collection.JavaConverters._
+
+/**
+ * The core part of the `\$search` pipeline stage of an aggregation pipeline.
+ * `SearchCollector`s allow returning metadata together with the matching search results.
+ * You may use the `$$SEARCH_META` variable, e.g., via [[Projections.computedSearchMeta]], to extract this metadata.
+ *
+ * @see [[https://www.mongodb.com/docs/atlas/atlas-search/operators-and-collectors/#collectors Search collectors]]
+ * @since 4.6
+ */
+object SearchCollector {
+
+  /**
+   * Returns a `SearchCollector` that groups results by values or ranges in the specified faceted fields and returns the count
+   * for each of those groups.
+   *
+   * @param operator A search operator to use.
+   * @param facets Non-empty facet definitions.
+   * @return The requested `SearchCollector`.
+   * @see [[https://www.mongodb.com/docs/atlas/atlas-search/facet/ facet collector]]
+   */
+  @Beta
+  def facet(operator: SearchOperator, facets: Iterable[_ <: SearchFacet]): FacetSearchCollector =
+    JSearchCollector.facet(operator, facets.asJava)
```

Comment:
After [fixing](https://github.com/mongodb/mongo-java-driver/pull/891/commits/50eee20077f35b308d9a9d800e63906034fe0299) another problem with Scala compiler, it now does not like having static methods in interfaces: _"SearchCollector.scala:46:27: Static methods in interface require -target:jvm-1.8"_. The error happens only for Scala 2.11 (`./gradlew -PscalaVersion=2.11.12 --stacktrace --info :bson-scala:test :driver-scala:test :driver-scala:integrationTest`). I tried specifying
```gradle
if(scalaVersion.startsWith("2.11")) {
    scalaCompileOptions.additionalParameters = [
            "-target:jvm-1.8"
    ]
}
```
in `build.gradle` inside the `tasks.withType(ScalaCompile) { ... }` block that we already have, but it did not help. @rozza, can you think of what I did wrong and what is the right way to pass `-target:jvm-1.8` to the Scala compiler?

Reply:
Investigating

Reply:
Thank you, Ross! I could have realized that it is worth checking if there is something in configuration of other modules that may overwrite parameters specified in `build.gradle`. Alas!

At first I tried moving the code you quoted from `drivers-scala/build.gradle` to `build.gradle`, but it resulted in failures of the `MacrosSpec.support tagged types in case classes` test. Because of this, I had to leave the code in `drivers-scala/build.gradle`, but change it so that it adds to parameters, instead of overwriting them. I added a similar piece of code to `build.gradle` in order to specify `-target:jvm-1.8`: it seems useful to allow static methods in interfaces for all our Scala code, not just that in the `drivers-scala` module.